From 41570c38c5e265025ae5cc901d396e013c3b94dd Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Sat, 11 Oct 2014 14:17:59 +0800
Subject: [PATCH 151/164] atomisp2: keep the error code and do returning at
 the end of atomisp_vidioc_default()

BZ: 226170

At the beginning and end of atomisp_videoc_default, there are some code to
control the mutex lock and unlock, therefore this patch changes all the code
in the middle to make the function returns at the end for easier control of
mutex.

Change-Id: If118127f076e9ac831229234da03d9abb945755c
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-8929
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://android.intel.com:443/286874
Reviewed-on: https://icggerrit.ir.intel.com/29107
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_ioctl.c    |   15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 583b0d1..23b82d9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -2734,13 +2734,14 @@ static long atomisp_vidioc_default(struct file *file, void *fh,
 
 	case ATOMISP_IOC_G_MOTOR_PRIV_INT_DATA:
 		if (isp->inputs[asd->input_curr].motor)
-			return v4l2_subdev_call(
+			err = v4l2_subdev_call(
 					isp->inputs[asd->input_curr].motor,
 					core, ioctl, cmd, arg);
 		else
-			return v4l2_subdev_call(
+			err = v4l2_subdev_call(
 					isp->inputs[asd->input_curr].camera,
 					core, ioctl, cmd, arg);
+		break;
 
 	case ATOMISP_IOC_S_EXPOSURE:
 	case ATOMISP_IOC_G_SENSOR_CALIBRATION_GROUP:
@@ -2749,8 +2750,9 @@ static long atomisp_vidioc_default(struct file *file, void *fh,
 	case ATOMISP_IOC_S_SENSOR_AE_BRACKETING_MODE:
 	case ATOMISP_IOC_G_SENSOR_AE_BRACKETING_MODE:
 	case ATOMISP_IOC_S_SENSOR_AE_BRACKETING_LUT:
-		return v4l2_subdev_call(isp->inputs[asd->input_curr].camera,
+		err = v4l2_subdev_call(isp->inputs[asd->input_curr].camera,
 					core, ioctl, cmd, arg);
+		break;
 
 	case ATOMISP_IOC_ACC_LOAD:
 		err = atomisp_acc_load(isp, arg);
@@ -2810,8 +2812,9 @@ static long atomisp_vidioc_default(struct file *file, void *fh,
 		err = atomisp_get_metadata_by_type(asd, 0, arg);
 		break;
 	case ATOMISP_IOC_EXT_ISP_CTRL:
-		return v4l2_subdev_call(isp->inputs[asd->input_curr].camera,
+		err = v4l2_subdev_call(isp->inputs[asd->input_curr].camera,
 					core, ioctl, cmd, arg);
+		break;
 	case ATOMISP_IOC_EXP_ID_UNLOCK:
 		err = atomisp_exp_id_unlock(asd, arg);
 		break;
@@ -2847,8 +2850,8 @@ static long atomisp_vidioc_default(struct file *file, void *fh,
 		err = atomisp_get_effective_res(asd, arg);
 		break;
 	default:
-		rt_mutex_unlock(&isp->mutex);
-		return -EINVAL;
+		err = -EINVAL;
+		break;
 	}
 
 	switch (cmd) {
-- 
1.7.9.5

